File: binance4j-wallet/src/main/java/com/binance4j/wallet/snapshot/futures/FuturesAccountSnapshot.java
package com.binance4j.wallet.snapshot.futures;

import com.binance4j.wallet.snapshot.Snapshot;
import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

/**
 * A daily FUTURES account snapshot
 */
@Data
@JsonAutoDetect(fieldVisibility = Visibility.ANY)
@JsonIgnoreProperties(ignoreUnknown = true)
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public class FuturesAccountSnapshot extends Snapshot {
	/**
	 * The account snapshot data
	 */
	private FuturesSnapshotData data;
}

File: wallet/src/main/kotlin/com/binance4j/wallet/dto/FuturesAccountSnapshot.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.wallet.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * A daily FUTURES account snapshot.
 *
 * @property type Type.
 * @property updateTime Update time in ms.
 * @property data Data.
 */
@ApiModel("A daily FUTURES account snapshot.")
data class FuturesAccountSnapshot(
    @ApiModelProperty("Type.") override @JsonProperty("type") var type: String? = null,
    @ApiModelProperty("Update time in ms.") override @JsonProperty("updateTime") var updateTime: Long? = null,
    @ApiModelProperty("Data.") @JsonProperty("data") var data: FuturesSnapshotData? = null
) : Snapshot
